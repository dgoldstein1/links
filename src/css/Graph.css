.contentContainer {
  display: flex;
  flex-direction: column;
  align-items: stretch;
  justify-content: flex-start;
  flex: 1 1 100%;
}

.graidentBackground { 
  background: linear-gradient(to bottom, orange, transparent 100%);
 }