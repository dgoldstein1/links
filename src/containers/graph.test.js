import React from "react";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import Graph from "./graph";

const mockStore = configureStore([]);

describe("Graph", () => {
  const defaultState = {
    graph: {
      graph: {
        nodes: [],
      },
      layout: "cluster",
      selectedNode: {
        node: {
          label: "test",
          id: 1,
        },
      },
    },
  };
  describe("matches snapshots", () => {
    let testTable = [
      {
        name: "cluster",
        state: defaultState,
      },
      {
        name: "loading",
        state: {
          ...defaultState,
          graph: {
            loading: true,
            ...defaultState.graph,
          },
        },
      },
      {
        name: "error",
        state: {
          ...defaultState,
          graph: {
            error: "an error",
            ...defaultState.graph,
          },
        },
      },
      {
        name: "hierarchy",
        state: {
          ...defaultState,
          graph: {
            ...defaultState.graph,
            layout: "hierarchy",
          },
        },
      },
    ];

    testTable.forEach((t) => {
      it(t.name, () => {
        let store = mockStore(t.state);
        let wrapper = renderer.create(
          <Provider store={store}>
            <Graph />
          </Provider>
        );
        expect(wrapper).toMatchSnapshot();
      });
    });
  });
  describe("sets graph error to undefined on click", () => {
    let store = mockStore({
      ...defaultState,
      graph: {
        error: "an error",
        ...defaultState.graph,
      },
    });
    let wrapper = renderer.create(
      <Provider store={store}>
        <Graph />
      </Provider>
    );
    // emulate click on button
    let t;
    renderer.act(() => {
      t = wrapper.root.findByProps({ id: "graph-error-card" }).props.onAction();
    });
    expect(t).not.toBeUndefined();
    expect(t).toMatchSnapshot();
  });
  describe("on node click", () => {
    let store = mockStore(defaultState);
    let wrapper = renderer.create(
      <Provider store={store}>
        <Graph />
      </Provider>
    );
    // emulate click on button
    renderer.act(() => {
      let e = {
        data: {
          node: {
            id: 123,
            label: "test",
          },
        },
      };
      wrapper.root.findByProps({ id: "sigma-graph" }).props.onClickNode(e);
    });
  });
});
