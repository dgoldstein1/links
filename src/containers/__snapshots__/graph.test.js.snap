// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
Object {
  "error": undefined,
  "type": "SET_GRAPH_ERROR",
}
`;

exports[`Graph matches snapshots cluster 1`] = `
<div
  className="contentContainer graidentBackground"
>
  <div
    className="hiddenScroll"
  >
    <div
      className="sigmaContainer"
    >
      <div
        style={
          Object {
            "height": "518px",
          }
        }
      />
    </div>
    <div
      className="selectedCardContainer"
    >
      <div
        className="row"
      >
        <div
          className="col-sm"
        >
          <div
            className="card fluid"
          >
            <div>
              <h1>
                <a
                  href="https://en.wikipedia.org/wiki/test"
                >
                  test
                </a>
              </h1>
              
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Graph matches snapshots error 1`] = `
<div
  className="contentContainer graidentBackground"
>
  <div
    className="errorCardContainer"
  >
    <div
      className="row"
    >
      <div
        className="col-sm"
      >
        <div
          className="card fluid warning"
        >
          <h1>
            warning
          </h1>
          <p>
            an error
          </p>
          <label
            className="drawer-close"
            htlmfor="drawer-control"
          />
          <br />
          <button
            onClick={[Function]}
          >
            close
          </button>
          <button>
            <a
              href="https://github.com/dgoldstein1/links/issues/new"
            >
              report bug
            </a>
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="hiddenScroll"
  >
    <div
      className="selectedCardContainer"
    />
  </div>
</div>
`;

exports[`Graph matches snapshots hierarchy 1`] = `
<div
  className="contentContainer graidentBackground"
>
  <div
    className="hiddenScroll"
  >
    <div
      className="sigmaContainer"
    >
      <div
        style={
          Object {
            "height": "518px",
          }
        }
      />
    </div>
    <div
      className="selectedCardContainer"
    >
      <div
        className="row"
      >
        <div
          className="col-sm"
        >
          <div
            className="card fluid"
          >
            <div>
              <h1>
                <a
                  href="https://en.wikipedia.org/wiki/test"
                >
                  test
                </a>
              </h1>
              
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Graph matches snapshots loading 1`] = `
<div
  className="contentContainer graidentBackground"
>
  <div
    className="spinner secondary"
  />
  <div
    className="hiddenScroll"
  >
    <div
      className="selectedCardContainer"
    >
      <div
        className="row"
      >
        <div
          className="col-sm"
        >
          <div
            className="card fluid"
          >
            <div>
              <h1>
                <a
                  href="https://en.wikipedia.org/wiki/test"
                >
                  test
                </a>
              </h1>
              
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
